python/pyabacus/src/py_diago_david.hpp

```diff
@@ -132,7 +132,6 @@ class PyDiagoDavid
             const std::complex<double> *psi_in, 
             std::complex<double> *spsi_out, 
             const int nrow, 
-            const int npw, 
```

Comment:
should npw always equals nrow?

Reply:
They are equal in general since `spsi` is the corresponding block of `S` times `psi`. In current code only `cg` and `dav` method support spsi and `npw` always equals `nrow` in the implementation.

`cg` offers its own wrapper of sPsi as follows, where `nrow` and `npw` are the same.
```cpp
auto spsi_func = [this, hm](const ct::Tensor& psi_in, ct::Tensor& spsi_out) {
            const auto ndim = psi_in.shape().ndim();
            REQUIRES_OK(ndim <= 2, "dims of psi_in should be less than or equal to 2");

            if (this->use_uspp)
            {
                // Convert a Tensor object to a psi::Psi object
                hm->sPsi(psi_in.data<T>(),
                         spsi_out.data<T>(),
                         ndim == 1 ? psi_in.NumElements() : psi_in.shape().dim_size(1),
                         ndim == 1 ? psi_in.NumElements() : psi_in.shape().dim_size(1),
                         ndim == 1 ? 1 : psi_in.shape().dim_size(0));
            }
            else
            // else simply do copy, i.e. S=I
              
```
And in `david`, the arguments are always `dim, dim, 1`.
```cpp
spsi_func(psi_in + m*ld_psi,&this->spsi[m*dim],dim,dim,1);
spsi_func(basis + dim*m, &this->spsi[m * dim], dim, dim, 1);
spsi_func(basis + dim*(nbase + m), &spsi[(nbase + m) * dim], dim, dim, 1);
spsi_func(basis + dim*(nbase + m), &spsi[(nbase + m) * dim], dim, dim, 1);
```
So it's better to change the wrapper function of `sPsi` -- to take the same leading dimension of psi and spsi. Thus it'll be consistent with Matrix-Matrix operators and `hpsi_func`.

In case npw is different from nrow in the future, we should provide a new wrapper or directly call original `sPsi`, while now there is no need to keep these two parameters that are always the same.